eo-maven-plugin/src/main/java/org/eolang/maven/SafeMojo.java

```diff
@@ -285,4 +298,15 @@ private static void closeTojos(final Closeable res) throws MojoFailureException
         }
     }
 
+    /**
+     * Executes mojo in limited time interval.
+     *
+     * @throws IOException If fails.
+     */
+    private void executeWithTimeout() throws IOException {
+        final Timeout timer = new Timeout(this.timeout);
+        timer.start();
+        this.exec();
+        timer.stop();
```

Comment:
@volodya-lombrozo not a major issue, but this code is not safe: `this.exec()` may finish at the same time with `timer.stop()`. The timer will try to do `Thread.interrupt()`, while the thread is already finished.

It would be better to let the thread decide when it's stopped. In other words, `timer.stop()` should be called from inside `this.exec()`.

Reply:
@yegor256 How is it possible? `timer.stop()` will always called after `this.exec()`. 
Maybe I've confused you, but the idea the next:
1. We create a separate thread that terminates the current thread when timeout is reached.
2. All the next commands `timer.start()`, `this.exec()` and `timer.stop()` are called in the same current thread (and actually the current thread decides when to stop timeout timer and it stops the timeout timer immediately after `this.exec()`).